File: 14_linked_list_get-node-value/get_node_value.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None

# def get_node_value(head, index):
#   i = 0
#   current = head
#   while current is not None:
#     if i == index:
#       return current.val
#     else:
#       i += 1
#       current = current.next
#   return None


def get_node_value(head, index):
  
  if head is None:
    return None
  if 0 == index:
    return head.val
  return get_node_value(head.next, index - 1)

  # Iterative => T: O(n) S: O(1)
  # Recursive => T: O(n) S: (On)

File: 14_linked_list_get-node-value/getNodeValue.js
const getNodeValue = (head, index) => {
  let count = 0
  let current = head
  while(current){
    if(count === index){
      return current.val
    }
    count++
    current = current.next
  }
  return null
};

const getNodeValue = (head, index) => {
  if(index === 0){
    return head.val
  }
  if(head === null){
    return null
  }
  return getNodeValue(head.next, index -1)
}